File: lab-1/src/main/java/com/evist0/labforwardlist/factory/impl/IntegerFactory.java
package com.evist0.labforwardlist.factory.impl;

import com.evist0.labforwardlist.factory.ObjectFactory;

import java.util.Comparator;

public class IntegerFactory implements ObjectFactory {
    @Override
    public Object parse(String input) {
        return Integer.parseInt(input);
    }

    @Override
    public String getObjectName() {
        return "Integer";
    }

    @Override
    public Comparator getComparator() {
        return Comparator.comparingInt(o -> (Integer) o);
    }
}

File: lab-2/src/main/scala/com/hamaddo/labforwardlist/factory/impl/IntegerFactory.scala
package com.hamaddo.labforwardlist.factory.impl

import com.hamaddo.labforwardlist.factory.ObjectFactory

import java.util.Comparator

private class IntegerFactory extends ObjectFactory {
  override def parse(input: String): Integer = {
    Integer.parseInt(input)
  }

  override def getObjectName: String = {
    "Integer"
  }

  override def getComparator: Comparator[Object] = {
    Comparator.comparingInt(o => o.asInstanceOf[Integer])
  }
}
